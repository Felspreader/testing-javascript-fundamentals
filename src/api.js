export default {
  isAuthenticated(user) {
    if (!user) {
      throw new Error('User is undefined')
    }
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        Math.random() > 0.5 ? resolve(true) : reject(false)
      }, 1000)
    })
  },
  submitOrder(user, payload) {
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        Math.random() > 0.5
          ? resolve(`/user/${user.id}/confirmation`)
          : reject('Something went wrong')
      }, 1000)
    })
  },
}
