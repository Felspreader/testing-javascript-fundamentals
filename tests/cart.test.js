import createCart from '../src/cart'
import Item from '../src/item'
import createItem from '../src/factories/item-factory'
import api from '../src/api'

let cart
beforeEach(() => {
  cart = createCart()
})

it('can add item', () => {
  const item = new Item({ name: 'Game Boy' })

  cart.add(item)

  expect(cart.items).toHaveLength(1)
  expect(cart.items[0].name).toBe(item.name)
  expect(cart.items[0]).toHaveProperty('qty')
  expect(cart.items[0].qty).toBe(1)
})

it('increases qty when adding an already added item', () => {
  const item = new Item({ name: 'Game Boy' })

  cart.add(item)
  cart.add(item)

  expect(cart.items).toHaveLength(1)
  expect(cart.items[0].qty).toBe(2)
})

it('throws if trying to remove when item does not exists', () => {
  expect(() => cart.remove(123)).toThrowError(
    'Item 123 does not exist in the cart.'
  )
})

it('removes existing item', () => {
  const item = new Item({ name: 'Game Boy' })

  cart.add(item)
  cart.remove(item.id)

  expect(cart.items).toEqual([])
})
