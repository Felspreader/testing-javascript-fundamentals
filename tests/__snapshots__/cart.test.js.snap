// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "add": [Function],
  "checkout": [Function],
  "coupon": null,
  "itemCount": [Function],
  "items": Array [],
  "orderTotal": [Function],
  "remove": [Function],
}
`;
